RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h

```diff
@@ -11,40 +15,41 @@ class SeedingHitSet {
 
   static ConstRecHitPointer nullPtr() { return nullptr; }
 
-  SeedingHitSet() { theRecHits[0] = theRecHits[1] = theRecHits[2] = theRecHits[3] = nullptr; }
+  SeedingHitSet() = default;
+
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two) : theRecHits({one, two}) { setSize(); }
 
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two)
-  // : theRecHits{{one,two,ConstRecHitPointer()}}
-  {
-    theRecHits[0] = one;
-    theRecHits[1] = two;
-    theRecHits[2] = theRecHits[3] = nullptr;
-  }
   SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three)
-  // : theRecHits{{one,two,three}},
-  {
-    theRecHits[0] = one;
-    theRecHits[1] = two;
-    theRecHits[2] = three;
-    theRecHits[3] = nullptr;
+      : theRecHits({one, two, three}) {
+    setSize();
   }
 
-  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three, ConstRecHitPointer four) {
-    theRecHits[0] = one;
-    theRecHits[1] = two;
-    theRecHits[2] = three;
-    theRecHits[3] = four;
+  SeedingHitSet(ConstRecHitPointer one, ConstRecHitPointer two, ConstRecHitPointer three, ConstRecHitPointer four)
+      : theRecHits({one, two, three, four}) {
+    setSize();
   }
 
-  ConstRecHitPointer const *data() const { return theRecHits; }
+  SeedingHitSet(std::vector<ConstRecHitPointer> &hits) : theRecHits(hits) { setSize(); }
+
+  virtual ~SeedingHitSet() { std::vector<ConstRecHitPointer>().swap(theRecHits); }
```

Comment:
is this non-default implementation needed?
Also, why `virtual` (I do not see any virtual methods)?